bin/elasticurl_cpp/main.cpp

```diff
@@ -0,0 +1,552 @@
+                          << aws_error_debug_str(tlsCtxOptions.LastError()) << std::endl;
+                exit(1);
+            }
+        }
+
+        if (appCtx.CaPath || appCtx.CaCert)
+        {
+            if (!tlsCtxOptions.OverrideDefaultTrustStore(appCtx.CaPath, appCtx.CaCert))
+            {
+                std::cerr << "Failed to load " << appCtx.CaPath << " and " << appCtx.CaCert << " with error "
+                          << aws_error_debug_str(tlsCtxOptions.LastError()) << std::endl;
+                exit(1);
+            }
+        }
+        if (appCtx.insecure)
+        {
+            tlsCtxOptions.SetVerifyPeer(false);
+        }
+
+        tlsContext = Io::TlsContext(tlsCtxOptions, Io::TlsMode::CLIENT, allocator);
+
+        tlsConnectionOptions = tlsContext.NewConnectionOptions();
+
+        if (!tlsConnectionOptions.SetServerName(hostName))
+        {
+            std::cerr << "Failed to set servername with error " << aws_error_debug_str(tlsConnectionOptions.LastError())
+                      << std::endl;
+            exit(1);
+        }
+        if (!tlsConnectionOptions.SetAlpnList(appCtx.alpn))
+        {
+            std::cerr << "Failed to load alpn list with error " << aws_error_debug_str(tlsConnectionOptions.LastError())
+                      << std::endl;
+            exit(1);
+        }
+    }
+    else
+    {
+        if (appCtx.RequiredHttpVersion == Http::HttpVersion::Http2)
+        {
+            std::cerr << "Error, we don't support h2c, please use TLS for HTTP/2 connection" << std::endl;
+            exit(1);
+        }
+        port = 80;
+        if (appCtx.uri.GetPort())
+        {
+            port = appCtx.uri.GetPort();
+        }
+    }
+
+    Io::SocketOptions socketOptions;
+    socketOptions.SetConnectTimeoutMs(appCtx.ConnectTimeout);
+
+    Io::EventLoopGroup eventLoopGroup(0, allocator);
+    if (!eventLoopGroup)
+    {
+        std::cerr << "Failed to create evenloop group with error " << aws_error_debug_str(eventLoopGroup.LastError())
+                  << std::endl;
+        exit(1);
+    }
+
+    Io::DefaultHostResolver defaultHostResolver(eventLoopGroup, 8, 30, allocator);
+    if (!defaultHostResolver)
+    {
+        std::cerr << "Failed to create host resolver with error "
+                  << aws_error_debug_str(defaultHostResolver.LastError()) << std::endl;
+        exit(1);
+    }
+
+    Io::ClientBootstrap clientBootstrap(eventLoopGroup, defaultHostResolver, allocator);
+    if (!clientBootstrap)
+    {
+        std::cerr << "Failed to create client bootstrap with error " << aws_error_debug_str(clientBootstrap.LastError())
+                  << std::endl;
+        exit(1);
+    }
+    clientBootstrap.EnableBlockingShutdown();
+
+    std::promise<std::shared_ptr<Http::HttpClientConnection>> connectionPromise;
+    std::promise<void> shutdownPromise;
+
+    auto onConnectionSetup = [&](const std::shared_ptr<Http::HttpClientConnection> &newConnection, int errorCode) {
```

Comment:
Sorry, I am not sure what's not necessary here.

Reply:
David's saying you should be explicit and name the exact variables used in the lambda capture, like:
`[&appCtx, &connectionPromise]` instead of just `[&]`

Then he was warning that clang will yell at you if you list more variables than are actually used